homestar-wasm/README.md

```diff
@@ -23,15 +23,1054 @@
+
+  ```wit
+  export fn: func(a: s32) -> s32;
+  ```
+
+  Given an JSON input for this function:
+
+  ```json
+  {
+    "args": [1]
+  }
+  ```
+
+  `1` is converted into an `Ipld::Integer`, which is then translated and
+  passed into `fn` as an integer argument (`s32`).
+
+  **Note**: However, If the input argument to the WIT interface is a `float`
+  type, but the incoming value is an `Ipld::Integer`, then the IPLD value will
+  be cast to a `float`, and remain as one for the rest of the computation. This
+  is to avoid weird JavaScript behavior around `1.0 === 1` for example.
+
+- **WIT to IPLD Translation**:
+
+  Conversely, when an integer value (not a float) is returned from a WIT
+  function, it can be translated back into an `Ipld::Integer`.
+
+**IPLD Schema Definitions**:
+
+```ipldschme
+type IPLDIntegerAsWit union {
+  | U8        int
+  | U16       int
+  | U32       int
+  | U64       int
+  | S8        int
+  | S16       int
+  | S32       int
+  | S64       int
+  | Float32In int
+  | Float64In int
+} representation kinded
+
+type WitAsIpldInteger union {
+  | U8          int
+  | U16         int
+  | U32         int
+  | U64         int
+  | S8          int
+  | S16         int
+  | S32         int
+  | S64         int
+  | Float32Out  float
+  | Float64Out  float
+} representation kinded
+```
+
+#### Floats
+
+This section outlines the translation process between IPLD float values
+(`Ipld::Float`) and [WIT `float` runtime values][wit-val].
+
+The [Component Model][wit] supports these Float types:
+
+```ebnf
+ty ::= 'float32' | 'float64'
+```
+
+- **IPLD to WIT Translation**:
+
+  When a WIT function expects a float input, an `Ipld::Float` value is
+  mapped to a float WIT runtime value. Casting is done to convert from `f32` to
+  `f64` if necessary.
+
+  **Example**: Consider a WIT function defined as follows:
+
+  ```wit
+  export fn: func(a: f64) -> f64;
+  ```
+
+  Given an JSON input for this function:
+
+  ```json
+  {
+    "args": [1.0]
+  }
+  ```
+
+  `1.0` is converted into an `Ipld::Float`, which is then translated and
+  passed into `fn` as a float argument (`f64`).
+
+- **WIT to IPLD Translation**:
+
+  Conversely, when a `float32` or `float64` value is returned from a WIT
+  function, it can be translated back into an `Ipld::Float`.
+
+  **Note**: In converting from `float32` to `float64`, the latter of which is
+  the default precision for [IPLD][ipld-float], precision will be lost.
+  **The interpreter will use decimal precision in this conversion**.
+
+**IPLD Schema Definitions**:
+
+```ipldsch
+type IPLDFloatAsWit union {
+  | Float32 float
+  | Float64 float
+} representation kinded
+
+type WitAsIpldFloat union {
+  | Float32 float
+  | Float64 float
+} representation kinded
+```
+
+#### Strings
+
+This section outlines the translation process between IPLD string values
+(`Ipld::String`) and varying [WIT runtime values][wit-val]. A `Ipld::String` value can be
```

Comment:
```suggestion
(`Ipld::String`) and various [WIT runtime values][wit-val]. A `Ipld::String` value can be
```
Possible change, "various" seems better to me here. (Same applies to repeated instances below.)